src/aragonjs-wrapper.js

```diff
@@ -19,9 +21,94 @@ const appSrc = (app = {}, gateway = ipfsDefaultConf.gateway) => {
 // Filter out apps without UI and add an appSrc property
 const prepareFrontendApps = (apps, gateway) =>
   apps
-    .filter(app => app.content && app.appId !== appIds['Vault'])
+    .map(app => ({ ...(appDefaults[app.appId] || {}), ...app }))
+    .filter(app => app && app['short_url'] && app.appId !== appIds['Vault'])
     .map(app => ({ ...app, appSrc: appSrc(app, gateway) }))
 
+// Keep polling accounts to notice when the user unlocks their wallet
+const pollAccounts = (web3, onAccounts) => {
+  if (!web3) {
+    return
+  }
+  web3.eth.getAccounts((err, accounts) => {
+    if (!err) {
+      onAccounts(accounts || [])
+    }
+    setTimeout(pollAccounts, ACCOUNTS_POLL_EVERY)
+  })
+}
+
+// Subscribe to wrapper's observables
+const subscribe = (
+  wrapper,
+  { onApps, onForwarders, onTransaction },
+  { ipfsConf }
+) => {
+  const { apps, forwarders, transactions } = wrapper
+  const workerSubscriptionPool = new WorkerSubscriptionPool()
+
+  const subscriptions = {
+    apps: apps.subscribe(apps => {
+      const frontendApps = prepareFrontendApps(apps, ipfsConf.gateway)
+      onApps(frontendApps, apps)
+    }),
+    connectedApp: null,
+    connectedWorkers: workerSubscriptionPool,
+    forwarders: forwarders.subscribe(onForwarders),
+    transactions: transactions.subscribe(onTransaction),
+    workers: apps.subscribe(apps => {
+      // Asynchronously launch webworkers for each new app that has a background
+      // script defined
+      apps
+        .map(app => ({ ...(appDefaults[app.appId] || {}), ...app }))
+        .filter(app => app.script)
+        .filter(
+          ({ proxyAddress }) => !workerSubscriptionPool.hasWorker(proxyAddress)
+        )
+        .forEach(async app => {
+          const { name, proxyAddress, script } = app
+          const scriptUrl =
+            removeTrailingSlash(appSrc(app, ipfsConf.gateway)) + script
+          let workerUrl = ''
+          try {
+            // WebWorkers can only load scripts from the local origin, so we
+            // have to fetch the script as text and make a blob out of it
+            workerUrl = await getBlobUrl(scriptUrl)
+          } catch (e) {
+            console.error(
+              `Failed to load ${name}(${proxyAddress})'s script (${script}): `,
+              e
+            )
+            return
+          }
+
+          const worker = new Worker(workerUrl)
+          worker.addEventListener(
+            'error',
+            err =>
+              console.log(
+                `Error from worker for ${name}(${proxyAddress}):`,
+                err
+              ),
+            false
+          )
+
+          const provider = new providers.MessagePortMessage(worker)
+          workerSubscriptionPool.addWorker({
+            app,
+            worker,
+            subscription: wrapper.runApp(provider, proxyAddress).shutdown,
```

Comment:
This will probably be an issue later as `runApp` will also return more methods, such as `setContext`.

Reply:
I don't think we'll ever need to use `setContext()` for the webworkers, since they should be caching all the state regardless of context?